test/server/config_validation/server_test.cc

```diff
@@ -73,14 +77,41 @@ class RuntimeFeatureValidationServerTest : public ValidationServerTest {
 
   static const std::vector<std::string> getAllConfigFiles() {
     setupTestDirectory();
+    return {"runtime_config.yaml"};
+  }
+};
 
-    auto files = TestUtility::listFiles(ValidationServerTest::directory_, false);
+class JsonApplicationLogsValidationServerTest : public ValidationServerTest {
+public:
+  static void SetUpTestSuite() { // NOLINT(readability-identifier-naming)
+    setupTestDirectory();
+  }
 
-    // Strip directory part. options_ adds it for each test.
-    for (auto& file : files) {
-      file = file.substr(directory_.length() + 1);
-    }
-    return files;
+  static void setupTestDirectory() {
+    directory_ =
```

Comment:
I may be missing something here, but why is the `directory_` needed?

Reply:
It is used by the MockOptions object maintained in the ``ValidationServerTest`` above - that's the source directory that the yaml configs are loaded for the tests. It was already embedded in the current test pattern (before I added this change), so I followed the same pattern.